containers/tenant-namespace-operator/roles/tenantnamespace/tasks/main.yml

```diff
@@ -32,8 +32,6 @@
     name: "{{ meta.name }}-ingress-controller"
     namespace: "{{ meta.name }}-admin"
   register: ingressService
-  when:
-  - _miscscripts_pnnl_gov_tenantnamespace.status.loadBalancerIP is not defined
```

Comment:
Will this cause it to run every time? ansible's job steps are not very fast so I'd prefer skipping them if possible.

Reply:
I'm not sure how to make the upgrade system work without this.  I need some way to tell that we are doing an upgrade of the ingress controller before we do a helm upgrade.  I could use the helm_info task and find the current chart version instead of this but I would probably need some sort of semver compare function to compare the version from helm.

Reply:
I guess maybe the status of the custom resource could be used for this too.

Reply:
Anything I can think of would be much more complicated... May we go ahead and merge this, and put the check back in along with removal of the deletion code after a while?

Reply:
I've added changes to setup a status marker when we have finished upgrading the dependent ingress chart.  Hopefully this fixes the issue for the time being and we can remove some of the code related to this upgrade in the future.